sprint-mission-6.sql

```diff
@@ -0,0 +1,194 @@
+### 초급 문제
+
+-- 1. `orders` 테이블에서 모든 주문을 조회하세요.
+SELECT * FROM orders;
+
+-- 2. `orders`테이블에서 `id` 가 `423`인 주문을 조회하세요.
+SELECT * FROM orders WHERE id = 423;
+
+-- 3. `orders` 테이블에서 총 주문 건수를 `total_orders`라는 이름으로 구하세요.
+SELECT COUNT(o.id) AS total_orders
+FROM orders AS o;
+
+-- 4. `orders` 테이블에서 최신 순으로 주문을 조회하세요. (`date`, `time` 컬럼이 분리되어 있다는 점에 주의)
+SELECT * FROM orders
+ORDER BY date DESC;
+
+-- 5. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때, 첫 번째 페이지를 조회하세요.
+SELECT * FROM orders
+ORDER BY date DESC, time DESC
+LIMIT 10 OFFSET 0;
+
+-- 6. `orders` 테이블에서 오프셋 기반 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일 때 5번째 페이지를 조회하세요.
+SELECT * FROM orders
+ORDER BY date DESC, time DESC
+LIMIT 10 OFFSET 49;
+
+-- 7. `orders` 테이블에서 커서 페이지네이션된 목록을 조회합니다. 페이지 크기가 10이고 최신순일때, `id` 값을 기준으로 커서를 사용합시다. 커서의 값이 `42`일 때 다음 페이지를 조회하세요.
+SELECT * FROM orders 
+WHERE id <= 42
+ORDER BY date DESC, time DESC
+LIMIT 10;
+
+-- 8. `orders` 테이블에서 2025년 3월에 주문된 내역만 조회하세요.
+SELECT * FROM orders 
+WHERE date >= '2025-03-01' AND date <= '2025-03-31';
+
+-- 9. `orders` 테이블에서 2025년 3월 12일 오전에 주문된 내역만 조회하세요.
+SELECT * FROM orders 
+WHERE date = '2025-03-12' AND time < '12:00:00';
+
+-- 10. `pizza_types` 테이블에서 이름에 'Cheese' 혹은 'Chicken'이 포함된 피자 종류를 조회하세요. (대소문자를 구분합니다)
+SELECT id FROM pizza_types
+WHERE name LIKE '%Cheese%'
+OR name LIKE '%Chicken%';
+
+### 중급 문제
+
+-- 1. `order_details` 테이블에서 각 피자(`pizza_id`)별로 주문된 건 수(`order_id`)를 보여주세요.
+SELECT pizza_id, COUNT(order_id) AS order_id
+FROM order_details
+GROUP BY pizza_id;
+-- 2. `order_details` 테이블에서 각 피자(`pizza_id`)별로 총 주문 수량을 구하세요.
+SELECT pizza_id, SUM(quantity) AS total_pizza
+FROM order_details
+GROUP BY pizza_id;
+-- 3. `pizzas` 테이블에서 `price`의 크기가 20보다 큰 피자의 종류만 `order_details` 테이블에서 조회하세요. (힌트: 서브쿼리)
+SELECT o.pizza_id FROM order_details AS o
+JOIN pizzas AS p
+ON o.pizza_id = p.id
+WHERE p.price > 20
+GROUP BY o.pizza_id;
+-- 4. `orders` 테이블에서 각 날짜별 총 주문 건수를 `order_count` 라는 이름으로 계산하고, 하루 총 주문 건수가 80건 이상인 날짜만 조회한 뒤, 주문 건수가 많은 순서대로 정렬하세요.
+SELECT * FROM order_details
+WHERE pizza_id IN(SELECT id FROM pizzas WHERE price > 20);
+-- 5. `order_details` 테이블에서 피자별(`pizza_id`) 총 주문 수량이 10개 이상인 피자만 조회하고, 총 주문 수량 기준으로 내림차순 정렬하세요.
+SELECT pizza_id, SUM(quantity) AS total_pizza
+FROM order_details
+GROUP BY pizza_id
+HAVING SUM(quantity) >= 10
+ORDER BY total_pizza DESC;
+-- 6. `order_details` 테이블에서 피자별 총 수익을 `total_revenue` 라는 이름으로 구하세요. (수익 = `quantity * price`)
+SELECT od.pizza_id, SUM(od.quantity) * p.price AS total_revenue
+FROM order_details AS od
+JOIN pizzas AS p
+ON od.pizza_id = p.id
+GROUP BY od.pizza_id, p.price;
```

Comment:
```suggestion
SELECT od.pizza_id,
       SUM(od.quantity * p.price) AS total_revenue
FROM order_details od
JOIN pizzas p ON p.id = od.pizza_id
GROUP BY od.pizza_id;
```
price로 그루핑을 하지 않고 "피자별"이라고 했으므로 pizza_id로 해야겠죠?
끄리고 계산식도 SUM(quantity * price)로 해야 좀 더 명확합니다.